src/Compiler/Microsoft.AspNetCore.Razor.Language/src/MetadataCollection.cs

```diff
@@ -69,7 +72,7 @@ public static MetadataCollection Create(params KeyValuePair<string, string?>[] p
             _ => new FourOrMoreItems(pairs),
         };
 
-    public static MetadataCollection Create(IReadOnlyList<KeyValuePair<string, string?>> pairs)
+    public static MetadataCollection Create(ImmutableArray<KeyValuePair<string, string?>> pairs)
```

Comment:
Curious: can you avoid the repetition here with the following signature:

```csharp
public static MetadataCollection Create<T>(T pairs)
  where T : IReadOnlyList<KeyValuePair<string, string?>>
{
   ... 
}
```

Reply:
interesting idea. I'll give it a try.

Reply:
Very nice! It works well. Thanks!